### ISSUE TITLE
Windows Install issues

### ISSUE THREAD

**Issue by Bassmann** [novoid/filetags#5]
I wanted to try your wonderful tagging idea using your scripts but encounter the following issues:
1. First I tried the "Ubuntu on Windows" environment which leads to this error:
Collecting filetags
  Using cached filetags-2017.12.30.tar.gz
Requirement already satisfied: colorama in /usr/local/lib/python2.7/dist-packages (from filetags)
Collecting importlib (from filetags)
  Using cached importlib-1.0.4.zip
Collecting logging (from filetags)
  Using cached logging-0.4.9.6.tar.gz
Collecting operator (from filetags)
  Could not find a version that satisfies the requirement operator (from filetags) (from versions: )
No matching distribution found for operator (from filetags)

It's pip 9.0.1 (python 2.7)

2. I cloned the git repo. Calling filetags.py again from "Ubuntu on Windows" leads to
Traceback (most recent call last):
  File "software/git-repos/filetags/filetags.py", line 129, in <module>
    :version: " + PROG_VERSION_DATE + "\n·\n"
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc2 in position 1: ordinal not in range(128)
which seems to indicate a Unicode issue. 

3. I installed python 3.6.4 for Windows and called filetags.py leading to
Could not find Python module "readline".
Please install it, e.g., with "sudo pip install readline".
Doing this leads to 
Collecting readline
  Downloading readline-6.2.4.1.tar.gz (2.3MB)
    Complete output from command python setup.py egg_info:
    error: this module is not meant to work on Windows 

4. I tried another route using docker:
I created a python 2.7 docker container which clones this repo and then runs filetags.py --help
This leads to
Traceback (most recent call last):
  File "filetags/filetags.py", line 129, in <module>
    :version: " + PROG_VERSION_DATE + "\n·\n"
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc2 in position 1: ordinal not in range(128)
which is the same error as in 2)

Any idea what to do?

**Comment by novoid** [comment]
Sorry for your problems. Windows is a hard nut to crack.

I myself am using filetags on Windows 10 without Ubuntu on Windows because I included filetags to the Windows Explorer and FreeCommander.

Please stick to Python 3 since Python 2 is obsolete from filetags perspective. You can install Python 2 in parallel but all libraries for filetags have to be installed for Python 3. I recommend using pip3 to install the libs.

My `filetags.bat` looks like this:
```
C:\Python36\python.exe C:\Users\karl.voit\src\filetags\filetags.py -i %*
REM set /p DUMMY=Hit ENTER to continue...
```

The thing with readline is that does not work with Windows but I am not an expert for that. 

You can try installing `pyreadline` as [comments on this page](http://richardt.name/blog/setting-up-ipython-notebook-on-windows/) suggest. As far as I remember I was able to overcome the same error using this library. The next thing which comes to mind is installing `pywin32`.

Please report back and I'll adapt the Readme.

**Comment by Bassmann** [comment]
Thanks for the feedback. Unfortunately still no success.

Now I uninstalled all python versions on my PC and installed the latest x64 version of python (3.6.4)

Still different errors occur:

1. pip3 install filetags leads to
  ```
Downloading importlib-1.0.4.zip
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "c:\users\<user>\appdata\local\programs\python\python36\lib\site-packages\setuptools\__init__.py", line 5, in <module>
        import distutils.core
      File "c:\users\<user>\appdata\local\programs\python\python36\lib\distutils\core.py", line 16, in <module>
        from distutils.dist import Distribution
      File "c:\users\<user>\appdata\local\programs\python\python36\lib\distutils\dist.py", line 19, in <module>
        from distutils.util import check_environ, strtobool, rfc822_escape
      File "c:\users\<user>\appdata\local\programs\python\python36\lib\distutils\util.py", line 9, in <module>
        import importlib.util
    ModuleNotFoundError: No module named 'importlib.util'
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in C:\Users\<user>\AppData\Local\Temp\pip-build-oqtqvbuq\importlib\
```
2. Running python filetags.py --version complained about
  - readline (I did as per your recommendation `pip3 install pyreadline`)
  - clint (`pip3 install clint`)
  - colorama (`pip3 install colorama`)
and finally
```
Traceback (most recent call last):
  File "filetags.py", line 63, in <module>
    import win32com.client
ModuleNotFoundError: No module named 'win32com'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "filetags.py", line 65, in <module>
    print("Could not find Python module \"" + library + "\".\nPlease install it, e.g., with \"sudo pip install " + library + "\".")
NameError: name 'library' is not defined
```
That's not found via `pip3 install win32com`.

Calling pythonw.exe instead doesnt give any errors but neither any output.

**Comment by novoid** [comment]
Try installing `pywin32` as well.

**Comment by sebastian-kuebeck** [comment]
I haven tested it with filetagd but I'd try virtualenv or miniconda to isolate python installations from each other.
https://virtualenv.pypa.io/en/stable/
https://conda.io/miniconda.html

**Comment by novoid** [comment]
Update: I found the problem and I'm onto it.

**Comment by novoid** [comment]
@Bassmann the issue should be fixed starting with the version "2018-01-31.1" - please try again and report back if there is anything else.

On Windows, you will need: `pip3 install filetags pypiwin32`

Thanks for pointing me to the issue I would not have found because I was only using the git repository.

**Comment by Bassmann** [comment]
Thanks for the immediate bugfix! Confirm it works now as described.
